pytorch_lightning/metrics/functional/reduction.py

```diff
@@ -22,3 +22,44 @@ def reduce(to_reduce: torch.Tensor, reduction: str) -> torch.Tensor:
     if reduction == 'sum':
         return torch.sum(to_reduce)
     raise ValueError('Reduction parameter unknown.')
+
+
+def class_reduce(num: torch.Tensor,
+                 denom: torch.Tensor,
+                 weights: torch.Tensor,
+                 class_reduction: str = 'none') -> torch.Tensor:
+    """
+    Function used to reduce classification metrics of the form `num / denom * weights`
+    For example for calculating standard accuracy the num would be number of
+    true positives pr class, denom would be the support pr class, and weights
+    would be a tensor of 1s
+
+    Args:
+        num: numerator tensor
+        decom: denomerator tensor
```

Comment:
Is this `denominator`?

Reply:
Yes :] please feel free to suggest changes to all typos